application/src/main/kotlin/net/corda/v5/application/flows/FlowEngine.kt

```diff
@@ -32,7 +32,7 @@ interface FlowEngine {
      * [Flow]s it communicated with. The subflow can be retried by catching this exception.
      */
     @Suspendable
-    fun <R> subFlow(subLogic: Flow<R>): R
+    fun <R> subFlow(subLogic: Subflow<R>): R
```

Comment:
Should we change to ` fun <R> subFlow(subLogic: SubFlow<R>, context: Map<String, String>): R` now too? Save ourselves breaking this API in another 2 minutes? Can throw not implemented if context is non-empty for now, but be good to settle the signature

Reply:
Definitely worth providing both on the API if there is a high chance of passing in `emptyMap` from a users perspective.

Reply:
I would much rather do this alongside other API changes in the context propagation epic rather than introducing it ad-hoc - we're going to have to revisit a lot of API when we come to that epic anyway.